sycl/include/sycl/accessor.hpp

```diff
@@ -1145,6 +1145,33 @@ class __SYCL_EBO __SYCL_SPECIAL_CLASS __SYCL_TYPE(accessor) accessor :
       : impl({}, detail::InitializedVal<AdjustedDim, range>::template get<0>(),
              detail::InitializedVal<AdjustedDim, range>::template get<0>()) {}
 
+  // implicit conversion between const / non-const types for read only accessors
+  template <typename DataType, int Dim, access::mode AccMode,
+            access::target AccTarget, access::placeholder Placeholder,
+            typename PropListT>
+  accessor(const accessor<DataType, Dim, AccMode, AccTarget, Placeholder,
+                          PropListT> &other,
```

Comment:
Do I understand correctly, that this technically allows conversions between accessors of different dimensions? We also don't check that `PropListT` is the same as `PropertyListT`

Reply:
Having different dimensions results in compiler error like below.
`no matching conversion for functional-style cast from 'accessor<int, 0, sycl::access::mode::read, sycl::target::device>' to 'accessor<int, 1, sycl::access::mode::read, sycl::target::device>'`
That should also be the case if the PropListT is not same type as  PropertyListT